src/nodes.coffee

```diff
@@ -266,7 +266,8 @@ exports.Base = class Base
   # For this node and all descendents, set the location data to `locationData`
   # if the location data is not already set.
   updateLocationDataIfMissing: (locationData) ->
-    return this if @locationData
+    return this if @locationData and not @forceUpdateLocation
```

Comment:
this `@forceUpdateLocation` is the workaround I described in the PR comment - it gets set inside `Value.add()` below to allow the subsequent call to `updateLocationDataIfMissing()` with (seemingly correct) updated location data that includes the location of the added property to flow through and set the `Value`'s `locationData`. This was the cleanest way I could see to allow updating the `locationData` for this one case